packages/providers/src/transaction-summary/get-transaction-summary.ts

```diff
@@ -0,0 +1,110 @@
+import { arrayify } from '@ethersproject/bytes';
+import { bn } from '@fuel-ts/math';
+import { TransactionCoder } from '@fuel-ts/transactions';
+
+import type { GqlGetTransactionsByOwnerQueryVariables } from '../__generated__/operations';
+import type Provider from '../provider';
+import type { TransactionRequest } from '../transaction-request';
+import type { TransactionResult } from '../transaction-response';
+import { processGqlReceipt } from '../transaction-response/utils';
+
+import { assembleTransactionSummary } from './assemble-transaction-summary';
+import type { TransactionSummary } from './types';
+
+export async function getTransactionSummary<TTransactionType = void>(
+  id: string,
+  provider: Provider
+): Promise<TransactionResult> {
+  const {
+    transaction: gqlTransaction,
+    chain: {
+      consensusParameters: { gasPerByte, gasPriceFactor },
+    },
+  } = await provider.operations.getTransactionWithReceipts({
+    transactionId: id,
+  });
+
+  if (!gqlTransaction) {
+    throw new Error('transaction not found');
+  }
+
+  const [decodedTransaction] = new TransactionCoder().decode(
+    arrayify(gqlTransaction.rawPayload),
+    0
+  );
+
+  const receipts = gqlTransaction.receipts?.map(processGqlReceipt) || [];
+
+  const transactionInfo = assembleTransactionSummary<TTransactionType>({
+    id: gqlTransaction.id,
+    gasPrice: bn(gqlTransaction.gasPrice),
+    receipts,
+    transaction: decodedTransaction,
+    transactionBytes: arrayify(gqlTransaction.rawPayload),
+    gqlTransactionStatus: gqlTransaction.status,
+    gasPerByte: bn(gasPerByte),
+    gasPriceFactor: bn(gasPriceFactor),
+  });
```

Comment:
what happens if it's a pending transaction (not processed yet in a block). it won't have receipts, but should work anyway and show the data we can proccess.

for example fees will not be shown because need receipts for that.

Reply:
Makes sense @LuizAsFight, nice catch :+1:

Reply:
@LuizAsFight Actually this seems to be working as expected already. 

What is the `graphql` answer when fetching a transaction that was not processed yet? 

If the transaction was not processed yet, is the `transaction` property inside of the `graphql` response going to be `null` / `undefined`? Or its status property will be of the type `SubmittedStatus`?

1. `null` or `undefined`: Since there is no data to process we can't do much else here besides throwing an error. We could brainstorm a more appropriate error message than `transaction not found`.
2. `SubmittedStatus`: The helper `assembleTransactionSummary` will process the transaction with the data that has in hand. The `fee` and `gasUsed` are going to default to `bn(0)`, since our helpers related to this calculation assign this value when there are no `Result` receipts to calculate the `gasUsed`, and of course, the `fee`.

Please let me know if I am missing something.

Reply:
@Torres-ssf it will be `SubmittedStatus`

I'm guessing it will work too, but would be nice to check this, maybe running a node that doesn't instantly process blocks

Reply:
@LuizAsFight I've just made some tests on a chain that does not instantly process blocks and could validate that this is working as expected